### PR TITLE
Include grade information

### PR DESCRIPTION
**Pull request by garemoko**
This PR includes all the changes in https://github.com/LearningLocker/Moodle-Log-Expander/issues/19 because I forgot to branch my repo before making that first PR. Once #19 merges, this PR should be easier to review. The changes specific to this one are in src/Events/AttemptEvent.php

I'll look at tests and docs now.

### CODE REVIEW

**Comment by garemoko** [comment]
@ryansmith94 c94906d is failing because I'm pulling data out of the Moodle database using the $DB global but $DB doesn't exist in the test framework. Any suggestions on how to get around this?

**Comment by ryansmith94** [comment]
@garemoko the `$DB` global should not be used in the events, only in the Repository. There is also a [Test Repository](https://github.com/LearningLocker/Moodle-Log-Expander/blob/master/tests/TestRepository.php) to fake fetching data from $DB for testing.

**Comment by garemoko** [comment]
@ryansmith94 thanks for the $DB tips. I'll work on that angle now, get this passing then add tests and get it passing again :-)

**Comment by ryansmith94** [comment]
No worries. Awesome :+1:

**Comment by garemoko** [comment]
@ryansmith94 this one is ready for review.

File: src/Repository.php [hunk 1]
```diff
@@ -69,6 +69,16 @@ public function readAttempt($id) {
     }
 
     /**
+     * Reads quiz grade metadata from the store with the given quiz id.
+     * @param String $id
+     * @return PhpObj
+     */
+    public function readGradeitems($id) {
```
Comment (ryansmith94) [inline src/Repository.php]: This needs to be changed to camel case (`readGradeItems`) to follow our code style.

File: src/Events/AttemptEvent.php [hunk 2]
```diff
@@ -9,9 +9,12 @@ class AttemptEvent extends Event {
      */
     public function read(array $opts) {
         $attempt = $this->repo->readAttempt($opts['objectid']);
+        $gradeitems = $this->repo->readGradeitems($attempt->quiz);
```
Comment (ryansmith94) [inline src/Events/AttemptEvent.php]: This needs to be changed to snake case (`$grade_items`) to follow our code style for variable names.

**Comment by ryansmith94** [comment]
Sorry I don't want to be picky about the code style stuff, but eventually I might add a linter to maintain the code style and I don't want to have to correct it all.

**Comment by garemoko** [comment]
@ryansmith94 happy to merge this now?

**Comment by ryansmith94** [comment]
Perfect! Thanks and sorry to be a pain. Hell yeah! :+1:

**Pull request merged**